### ISSUE TITLE
Can we remove support for XSPEC models from load_table_model

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#902]
In the CIAO 4.9 release I added the warning to the load_table_model function that support for XSPEC models would be removed ([*]). I'd like that time to be now...

[*] Why should it be removed? Well, the current code is very convoluted, and this would help slightly. I also believe it is confusing for users, since the table model interface is *very* basic whereas the XSPEC table model support is much-more complex, and so combining the two is confusing. I would say that the XSPEC table model support is more-closely related to the template model support than table model, but as I have essentially no experience in the template model code I can't say for sure. And I would not want the template model code to have special-case behavior for the XSPEC table models.

```
    def load_table_model(self, modelname, filename, method=sherpa.utils.linear_interp, *args, **kwargs):
        """Load tabular or image data and use it as a model component.
    
        .. note:: Deprecated in Sherpa 4.9
                  The new `load_xstable_model` routine should be used for
                  loading XSPEC table model files. Support for these files
                  will be removed from `load_table_model` in the next
                  release.
    
        A table model is defined on a grid of points which is
        interpolated onto the independent axis of the data set.  The
        model will have at least one parameter (the amplitude, or
        scaling factor to multiply the data by), but may have more
        (if X-Spec table models are used).
```

**Comment by DougBurke** [comment]
We have just got a helpdesk ticket from a user who was using load_table_model to try and read in a XSPEC table model but it failed with

```
IOErr: file 'mytorus_scatteredH400_v00.fits' does not appear to be ASCII
```

The actual problem reported by `load_xstable_model` is

```
ParameterErr: parameter foo.incang has a maximum of 87.1
```

but this message gets lost from all the work that `load_table_model` has to do to support all the different file options.

**Comment by anetasie** [comment]
I agree we should remove xspec tables. Can we have an error msgs pointing to a different function for people who still want to read it using `load_table_model`? Should we first include a deprecation warning? I'm worried about people who have been using it.
